diff -Nabur a/shim-15.8/shim.c b/shim-15.8/shim.c
--- a/shim-15.8/shim.c	2024-01-22 20:18:05.000000000 +0100
+++ b/shim-15.8/shim.c	2024-02-13 09:22:14.755599400 +0100
@@ -1307,13 +1307,13 @@
 		return EFI_SUCCESS;
 	}
 #endif
-
+#if 0
 	efi_status = parse_load_options(li);
 	if (EFI_ERROR(efi_status)) {
 		perror (L"Failed to get load options: %r\n", efi_status);
 		return efi_status;
 	}
-
+#endif
 	return EFI_SUCCESS;
 }
 
