diff -Nabur a/shim-16.1/shim.c b/shim-16.1/shim.c
--- a/shim-16.1/shim.c	2025-07-30 09:54:28.613297919 +0200
+++ b/shim-16.1/shim.c	2025-08-01 09:00:43.273067685 +0200
@@ -507,13 +507,13 @@
 		return EFI_SUCCESS;
 	}
 #endif
-
+#if 0
 	efi_status = parse_load_options(li);
 	if (EFI_ERROR(efi_status)) {
 		perror (L"Failed to get load options: %r\n", efi_status);
 		return efi_status;
 	}
-
+#endif
 	return EFI_SUCCESS;
 }
 
