diff -Nabur a/shim-15.6/shim.c b/shim-15.6/shim.c
--- a/shim-15.6/shim.c	2022-06-24 11:50:32.018238972 +0200
+++ b/shim-15.6/shim.c	2022-06-24 11:52:56.362482731 +0200
@@ -1265,13 +1265,13 @@
 		return EFI_SUCCESS;
 	}
 #endif
-
+#if 0
 	efi_status = parse_load_options(li);
 	if (EFI_ERROR(efi_status)) {
 		perror (L"Failed to get load options: %r\n", efi_status);
 		return efi_status;
 	}
-
+#endif
 	return EFI_SUCCESS;
 }
 
