diff -Nabur a/shim-15.4/shim.c b/shim-15.4/shim.c
--- a/shim-15.4/shim.c	2021-03-23 19:39:44.000000000 +0100
+++ b/shim-15.4/shim.c	2021-03-25 16:23:45.481195311 +0100
@@ -996,7 +996,7 @@
  * Protocol entry point. If secure boot is enabled, verify that the provided
  * buffer is signed with a trusted key.
  */
-EFI_STATUS shim_verify (void *buffer, UINT32 size)
+EFI_STATUS __attribute__((ms_abi)) shim_verify (void *buffer, UINT32 size)
 {
 	EFI_STATUS efi_status = EFI_SUCCESS;
 	PE_COFF_LOADER_IMAGE_CONTEXT context;
@@ -1041,7 +1041,7 @@
 	return efi_status;
 }
 
-static EFI_STATUS shim_hash (char *data, int datasize,
+static EFI_STATUS __attribute__((ms_abi)) shim_hash (char *data, int datasize,
 			     PE_COFF_LOADER_IMAGE_CONTEXT *context,
 			     UINT8 *sha256hash, UINT8 *sha1hash)
 {
@@ -1058,7 +1058,7 @@
 	return efi_status;
 }
 
-static EFI_STATUS shim_read_header(void *data, unsigned int datasize,
+static EFI_STATUS __attribute__((ms_abi)) shim_read_header(void *data, unsigned int datasize,
 				   PE_COFF_LOADER_IMAGE_CONTEXT *context)
 {
 	EFI_STATUS efi_status;
diff -Nabur a/shim-15.4/shim.h b/shim-15.4/shim.h
--- a/shim-15.4/shim.h	2021-03-23 19:39:44.000000000 +0100
+++ b/shim-15.4/shim.h	2021-03-25 16:22:53.189034476 +0100
@@ -217,9 +217,9 @@
 	);
 
 typedef struct _SHIM_LOCK {
-	EFI_SHIM_LOCK_VERIFY Verify;
-	EFI_SHIM_LOCK_HASH Hash;
-	EFI_SHIM_LOCK_CONTEXT Context;
+	__attribute__((ms_abi)) EFI_SHIM_LOCK_VERIFY Verify;
+	__attribute__((ms_abi)) EFI_SHIM_LOCK_HASH Hash;
+	__attribute__((ms_abi)) EFI_SHIM_LOCK_CONTEXT Context;
 } SHIM_LOCK;
 
 extern EFI_STATUS shim_init(void);
